test/operation-security.test.js

```diff
@@ -0,0 +1,125 @@
+const { linterForRule } = require('./utils');
+
+let linter;
+
+beforeAll(async () => {
+  linter = await linterForRule('az-operation-security');
+  return linter;
+});
+
+test('az-operation-security should find operations without security', () => {
+  const oasDoc = {
+    swagger: '2.0',
+    paths: {
+      '/test1': {
+        get: {
+          operationId: 'notNounVerb',
+        },
+        post: {
+          operationId: 'fooBarBaz',
+        },
+      },
+    },
+  };
+  return linter.run(oasDoc).then((results) => {
+    expect(results).toHaveLength(2);
+    expect(results[0].path.join('.')).toBe('paths./test1.get');
+    expect(results[1].path.join('.')).toBe('paths./test1.post');
+    results.forEach((result) => expect(result.message).toContain(
+      'Operation should have a security requirement.',
+    ));
+  });
+});
+
+test('az-operation-security should find operations without security', () => {
```

Comment:
Should the name of the test specify the document type that's it's targeting? As this has the same name as the above test.

Reply:
Agreed.  I wish jest allowed grouping of tests so that I could have a separate group for oas3 with the same descriptions, but AFAIK it does not.  I will fix.